ci-operator/step-registry/cucushift/hypershift-extended/external-oidc/enable/cucushift-hypershift-extended-external-oidc-enable-commands.sh

```diff
@@ -73,24 +74,33 @@ HOSTED_CLUSTER_RELEASE=$(yq-v4 'select(.kind == "HostedCluster") | .spec.release
-    if curl -sS https://raw.githubusercontent.com/openshift/api/refs/heads/release-"$HOSTED_CLUSTER_VERSION"/payload-manifests/featuregates/featureGate-Hypershift-Default.yaml | yq-v4 '.status.featureGates[].enabled' | grep -q ExternalOIDCWithUIDAndExtraClaimMappings; then
-        CREATED_CLAIM_MAPPINGS=$(oc get hc/"$HOSTED_CLUSTER_NAME" -o jsonpath='{.spec.configuration.authentication.oidcProviders[*].claimMappings}')
-        if jq '.uid' <<< "$CREATED_CLAIM_MAPPINGS" | grep -q testuid && jq -c '.extra' <<< "$CREATED_CLAIM_MAPPINGS" | grep -q 'bar.*foo'; then
-            echo "HostedCluster: External OIDC uid and extra settings are honored."
-        else
-            echo "$CREATED_CLAIM_MAPPINGS"
-            echo "HostedCluster: External OIDC uid and extra settings are not honored!"
-            exit 1
-        fi
-    fi
-fi
+# TODO: Uncomment this when OCPBUGS-57736 is fixed.
+# echo "Checking External OIDC uid and extra settings ..."
+# echo "First, checking ExternalOIDCWithUIDAndExtraClaimMappings featuregate ..."
+# if [[ $(awk "BEGIN {print ($HOSTED_CLUSTER_VERSION >= 4.18)}") == "1"  ]]; then
+#     # Once the ExternalOIDCWithUIDAndExtraClaimMappings feature PRs are merged and backported to 4.18, remove the `curl` line
+#     if curl -sS https://raw.githubusercontent.com/openshift/api/refs/heads/release-"$HOSTED_CLUSTER_VERSION"/payload-manifests/featuregates/featureGate-Hypershift-Default.yaml | yq-v4 '.status.featureGates[].enabled' | grep -q ExternalOIDCWithUIDAndExtraClaimMappings; then
+#         CREATED_CLAIM_MAPPINGS=$(oc get hc/"$HOSTED_CLUSTER_NAME" -o jsonpath='{.spec.configuration.authentication.oidcProviders[*].claimMappings}')
+#         if jq '.uid' <<< "$CREATED_CLAIM_MAPPINGS" | grep -q testuid && jq -c '.extra' <<< "$CREATED_CLAIM_MAPPINGS" | grep -q 'bar.*foo'; then
+#             echo "HostedCluster: External OIDC uid and extra settings are honored."
+#         else
+#             echo "$CREATED_CLAIM_MAPPINGS"
+#             echo "HostedCluster: External OIDC uid and extra settings are not honored!"
+#             exit 1
+#         fi
+#     fi
+# fi
 
-echo "Creating the console client secret"
-oc create secret generic "$CONSOLE_CLIENT_SECRET_NAME" -n clusters --from-literal=clientSecret="$CONSOLE_CLIENT_SECRET"
+if [ "${HYPERSHIFT_MANAGED_SERVICE:-}" = "ARO-HCP" ] && [ "$(awk "BEGIN {print ($HOSTED_CLUSTER_VERSION >= 4.20)}")" = "1" ]; then
+  echo "Creating an empty secret for the console client"
+  # The secret will be populated by the day2 operator in the hosted cluster.
+  oc create secret generic "$CONSOLE_CLIENT_SECRET_NAME" -n clusters
+  oc annotate secret "$CONSOLE_CLIENT_SECRET_NAME" -n clusters hypershift.openshift.io/hosted-cluster-sourced=true
+else
+  echo "Creating the console client secret"
+  oc create secret generic "$CONSOLE_CLIENT_SECRET_NAME" -n clusters --from-literal=clientSecret="$CONSOLE_CLIENT_SECRET"
+fi
```

Comment:
@mgencur , the original file's indentation uses 4 spaces.

Reply:
OK. Fixing